src/core/algorithms/od/od_verifier/partition.cpp

```diff
@@ -0,0 +1,57 @@
+#include "partition.h"
+
+#include <strings.h>
+#include <utility>
+#include <vector>
+
+namespace algos::od_verifier {
+
+std::vector<std::pair<int, int>> Partition::CommonViolationBySplit(model::ColumnIndex right) const {
```

Comment:
It is better to have `using X = std::pair<int, int>` for clarity of what std::pair<int, int> is.